packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -159,6 +164,17 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 				assert(knownSessionId.value === value.value, 0xa5a /* Mismatched SessionId */);
 			}
 		}
+
+		const audienceMembers = this.audience.getMembers();
+		const attendees = new Set(this.attendees.values());
+
+		for (const attendee of attendees) {
+			const connectionId = attendee.getConnectionId();
+			if (!audienceMembers.has(connectionId) && senderConnectionId !== connectionId) {
+				attendee.setDisconnected();
```

Comment:
Looks like we can get attendees that are announced as joined but are disconnected. We're not avoiding that?